### ISSUE TITLE
Error -9988 Steam closed

### ISSUE THREAD

**Issue by datatalking** [Uberi/speech_recognition#370]
Steps to reproduce
------------------

1. I am trying to use Speech_Recognition
2. When I run it from Pycharm or from my bash I generate errors.

Expected behavior
------------------

I expected the speech_recognition file to run, as it did in December 2017. 

Actual behavior
----------------

Bash Error
$ python converse.py 
Initializing...
Traceback (most recent call last):
  File "converse.py", line 116, in <module>
    listen(mic, recog)
NameError: name 'mic' is not defined

Pycharm Error
```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/owner/Library/Application Support/JetBrains/Toolbox/apps/PyCharm-P/ch-1/182.3911.3/PyCharm 2018.2 EAP.app/Contents/helpers/pydev/_pydev_bundle/pydev_umd.py", line 194, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "/Users/owner/Library/Application Support/JetBrains/Toolbox/apps/PyCharm-P/ch-1/182.3911.3/PyCharm 2018.2 EAP.app/Contents/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/owner/sbox/James/JamesFiles/jamesmk6_2/converse.py", line 115, in <module>
    initialize_mic()
  File "/Users/owner/sbox/James/JamesFiles/jamesmk6_2/converse.py", line 54, in initialize_mic
    recog.adjust_for_ambient_noise(mic)
  File "/Users/owner/ThirdParty/speech_recognition/speech_recognition/__init__.py", line 188, in __exit__
    self.stream.close()
  File "/Users/owner/ThirdParty/speech_recognition/speech_recognition/__init__.py", line 203, in close
    if not self.pyaudio_stream.is_stopped():
  File "/Users/owner/ThirdParty/venv/lib/python3.7/site-packages/pyaudio.py", line 543, in is_stopped
    return pa.is_stream_stopped(self._stream)
OSError: [Errno -9988] Stream closed
```

System information
------------------

(Delete all the statements that don't apply.)

My iMac is macOS High Sierra

My **Python version** is 3.7.0a2

My **Pip version** is 18

My **SpeechRecognition library version** is 3.6.3

My **PyAudio library version** is 0.2.11

My **microphones** are: ['Built-in Microphone', 'Built-in Input', 'Built-in Output'], bose in ear headphones

My **working microphones** are: 
python -c "import speech_recognition as sr;print(sr.Microphone.list_working_microphones())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: type object 'Microphone' has no attribute 'list_working_microphones'

Ummmm problem...

**Comment by datatalking** [comment]
This might be related to

Stream Closed Error #71
[https://github.com/Uberi/speech_recognition/issues/71](url)

**Comment by palikar** [comment]
This was fixed by #306 .
